modules/ITK/cli/statismo-build-gp-model-kernels.h

```diff
@@ -0,0 +1,89 @@
+#include <string>
+#include <vector>
+#include <map>
+
+#include <itkImage.h>
+#include <itkMesh.h>
+
+#include <boost/algorithm/string.hpp>
+
+#include "Kernels.h"
+#include "KernelCombinators.h"
+
+using namespace std;
+
+/*
+You can add your kernels here:
+	1. Either #include your kernel or put the class in here
+	2. Write a function in this file that creates your kernel (create it on the heap and NOT on the stack; it will be deleted, don't worry)
+	   Your function will receive a vector containing the kernel arguments (vector<string>). You have to parse them or thrown an itk::ExceptionObject if something goes wrong.
+	3. Add your kernel to the kernel map (do this in the function createKernelMap(); your kernel name has to be lowercase)
+	4. Document your kernel in the statismo-build-gp-model.md file. (Examples for kernels with multiple parameters are in there, but currently commented out)
+*/
+
+
+const unsigned Dimensions = 3;
+typedef itk::Mesh<float, Dimensions> DataTypeShape;
+typedef itk::Image< itk::Vector<float, Dimensions>, Dimensions > DataTypeDeformation;
+struct KernelContainer {
+    const statismo::ScalarValuedKernel<DataTypeShape::PointType>* (*createKernelShape)(vector<string> kernelArgs);
+    const statismo::ScalarValuedKernel<DataTypeDeformation::PointType>* (*createKernelDeformation)(vector<string> kernelArgs);
+};
+typedef map<string, KernelContainer> KernelMapType;
+KernelMapType kernelMap;
+
+
+template <class TPoint>
+class GaussianKernel : public statismo::ScalarValuedKernel<TPoint> {
+  public:
+    typedef typename  TPoint::CoordRepType CoordRepType;
+    typedef vnl_vector<CoordRepType> VectorType;
+
+    GaussianKernel(double sigma) : m_sigma(sigma), m_sigma2(sigma * sigma) {}
```

Comment:
would make sense to store 1 . / m_sigma2 (if it is going to be used extensively)